### PR TITLE
[MODEXPS-37] - Added new export types

### PR DESCRIPTION
**Pull request by Dmitriy-Butramyou**
## Purpose
Add new export types
 - BATCH_VOUCHER_EXPORT
 - EDIFACT_ORDERS_EXPORT
 - ORDERS_EXPORT
 - INVOICE_EXPORT

## Approach
- Added new export types.
- Wrote the migration.

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/resources/db/changelog/changes/db.changelog-4.1.0.xml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+  <changeSet id="MODEXPS-37@@cast exportType to varying" author="d.butramyou">
+    <sql dbms="postgresql">
+      CREATE CAST (character varying as ExportType) WITH INOUT AS IMPLICIT;
+    </sql>
+  </changeSet>
+
+  <changeSet id="MODEXPS-37@@rename column(s) which uses our enum type" author="d.butramyou">
+    <sql dbms="postgresql">
+      ALTER TABLE diku_mod_data_export_spring.job rename COLUMN type TO _type;
```
Comment (AndreiMPV) [inline src/main/resources/db/changelog/changes/db.changelog-4.1.0.xml:28]: Could you please remove hardcode of the schema name : "diku_mod_data_export_spring"

File: src/main/resources/db/changelog/changes/db.changelog-4.1.0.xml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  <changeSet id="MODEXPS-37@@rename the enum type you want to change" author="d.butramyou">
+    <sql dbms="postgresql">
+      ALTER TYPE ExportType rename TO _export_type;
+    </sql>
+  </changeSet>
+
+  <changeSet id="MODEXPS-37@@create new type" author="d.butramyou">
+    <sql dbms="postgresql">
+      CREATE TYPE ExportType AS enum ('CIRCULATION_LOG', 'BURSAR_FEES_FINES',
```
Comment (AndreiMPV) [inline src/main/resources/db/changelog/changes/db.changelog-4.1.0.xml:15]: Can we do that without table recreation from scratch with old and new ExportTypes?

**Review by AndreiMPV — CHANGES_REQUESTED** [review]

File: src/main/resources/db/changelog/changes/db.changelog-4.1.0.xml [hunk 3]
```diff
@@ -0,0 +1,62 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
```
Comment (AndreiMPV) [inline src/main/resources/db/changelog/changes/db.changelog-4.1.0.xml:1]: Can we replace this with SQL script?

**Review by AndreiMPV — CHANGES_REQUESTED** [review]

**Comment by Dmitriy-Butramyou** [comment]
Checked by the postman:
image
image

**Review by khandramai — APPROVED** [review]

**Review by AndreiMPV — APPROVED** [review]

**Pull request merged**